.github/workflows/build_and_test.yml

```diff
@@ -933,6 +933,46 @@ jobs:
       - name: Cleanup ECR folder
         run: rm -rf ~/.ecr
 
+  build-private-extensions:
+    runs-on: [ self-hosted, gen3, small ]
+    container:
+      image: 369495373322.dkr.ecr.eu-central-1.amazonaws.com/base:pinned
+      options: --init
+    needs: [ promote-images, tag ]
+    steps:
+      - name: Set PR's status to pending and request a remote CI test
+        run: |
+          COMMIT_SHA=${{ github.event.pull_request.head.sha }}
+          COMMIT_SHA=${COMMIT_SHA:-${{ github.sha }}}
+          REMOTE_REPO="${{ github.repository_owner }}/build-custom-extensions"
+
+          curl -f -X POST \
+          https://api.github.com/repos/${{ github.repository }}/statuses/$COMMIT_SHA \
+          -H "Accept: application/vnd.github.v3+json" \
+          --user "${{ secrets.CI_ACCESS_TOKEN }}" \
+          --data \
+            "{
+              \"state\": \"pending\",
+              \"context\": \"build-and-upload-extensions\",
+              \"description\": \"[$REMOTE_REPO] Remote CI job is about to start\"
+            }"
+
+          curl -f -X POST \
+          https://api.github.com/repos/$REMOTE_REPO/actions/workflows/build_and_upload_extensions.yml/dispatches \
+          -H "Accept: application/vnd.github.v3+json" \
+          --user "${{ secrets.CI_ACCESS_TOKEN }}" \
+          --data \
+            "{
+              \"ref\": \"main\",
+              \"inputs\": {
+                \"ci_job_name\": \"build-and-upload-extensions\",
+                \"commit_hash\": \"$COMMIT_SHA\",
+                \"remote_repo\": \"${{ github.repository }}\",
+                \"compute_image_tag\": \"${{ needs.tag.outputs.build-tag }}\",
+                \"remote_branch_name\": \"${{ github.ref_name }}\"
+              }
+            }"
```

Comment:
you could save yourself some heartache here if you wrap the json in single quotes. Then you don't have to escape the double quotes.

Reply:
And then while you're here, might as well indent the continuation of the curl command.

Reply:
Also, have you seen https://github.com/actions/github-script? Alternative solution if you don't want to keep curl

Reply:
thanks! I will use single quotes

Reply:
although for some reason in the other places where we do this kind of stuff it's escaped double quotes

Reply:
maybe for consistency I should either not change it or change in both places?
not sure if @bayandin has an opinion? (I think he wrote the other `curl` commands)

Reply:
We can also use `gh workflow run ...`.
Agree that this can be done in a separate PR.